controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -99,6 +99,26 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// Autoscaling specifies auto scaling behaviour for the MachinePools.
 	// +optional
 	Autoscaling *expinfrav1.RosaMachinePoolAutoScaling `json:"autoscaling,omitempty"`
+
+	// +kubebuilder:validation:Optional
+
+	// AdditionalTags are user-defined tags to be added on the AWS resources associated with the control plane.
+	AdditionalTags infrav1.Tags `json:"additionalTags,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^arn:aws[\\w-]*:kms:[\\w-]+:\\d{12}:key\\/(mrk-[0-9a-f]{32}$|[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$)')`, message="etcdEncryptionKMSArn must be a valid encryption key ARN"
+
+	// EtcdEncryptionKMSArn is the ARN of the KMS key used to encrypt etcd. The key itself needs to be
+	// created out-of-band by the user and tagged with `red-hat:true`.
+	EtcdEncryptionKMSArn string `json:"etcdEncryptionKMSArn,omitempty"`
+
+	// +kubebuilder:validation:Optional
+
+	// Private restricts master API endpoint and application routes to direct, private connectivity.
+	// Traffic to these endpoints will use AWS PrivateLink to have connectivity between VPCs, AWS services,
+	// and your on-premises networks without exposing your traffic to the public internet.
+	Private *bool `json:"private,omitempty"`
```

Comment:
why is this a pointer? pointers imply a third option if not set, but we only have 2 options.
This should just be a `bool` with a default to false.

Reply:
That's what the OCM API / ROSA sdk wants, I don't see a compelling reason to change that. A `*bool` allows distinguishing "unset" from "not private."

Reply:
What does unset mean in this context? If it's the same as `false`, we usually don't use pointers

Reply:
that is my question, what is unset? its either private=true or not=false, what happens when unset?

Reply:
We are a trivial passthrough to the OCM API. I don't see any reason why we should choose a different semantic than they do.

Reply:
By "trivial passthrough" I mean - we don't do any processing on this, we don't read the value, we don't do anything other than slap it into the OCM API call and call it a day.

Reply:
Can do the same with `&` before the field to OCM API

Reply:
No, we lose the unset semantic.